server/container/lifecycle-api/src/main/java/org/apache/james/lifecycle/api/Disposable.java

```diff
@@ -29,4 +39,111 @@
      * Dispose the object
      */
     void dispose();
+
+    abstract class LeakAware implements Disposable {
```

Comment:
Why grouping those new classes inside the Disposable interface and not extracting them separately?

Reply:
IMO it make sense from a semantic point of view : as a caller I would type `Disposeable.LeakAware` which is very explicit. IMO with separated class you loose such a reading.